components/core/src/clp/streaming_archive/ArchiveMetadata.cpp

```diff
@@ -22,15 +29,27 @@ ArchiveMetadata::ArchiveMetadata(
                          + sizeof(m_end_timestamp) + sizeof(m_compressed_size);
 }
 
-ArchiveMetadata::ArchiveMetadata(FileReader& file_reader) {
-    file_reader.read_numeric_value(m_archive_format_version, false);
-    file_reader.read_numeric_value(m_creator_id_len, false);
-    file_reader.read_string(m_creator_id_len, m_creator_id, false);
-    file_reader.read_numeric_value(m_creation_idx, false);
-    file_reader.read_numeric_value(m_uncompressed_size, false);
-    file_reader.read_numeric_value(m_compressed_size, false);
-    file_reader.read_numeric_value(m_begin_timestamp, false);
-    file_reader.read_numeric_value(m_end_timestamp, false);
+auto ArchiveMetadata::create_from_file(std::string_view path) -> ArchiveMetadata {
+    auto file_reader = FileReader(std::string(path));
```

Comment:
```suggestion
     FileReader file_reader{std::string(path)};
```

* Using `auto` in this case is more verbose and unnecessary since we need to explicitly specify the type anyway.
* We should also use brace init.